File: src/theme/typography.ts
/**
 * Typography presets for the MaaS app.
 */

import { Platform, TextStyle } from 'react-native';

export const Fonts = Platform.select({
  ios: {
    sans: 'system-ui' as const,
    serif: 'ui-serif' as const,
    rounded: 'ui-rounded' as const,
    mono: 'ui-monospace' as const,
  },
  default: {
    sans: 'normal' as const,
    serif: 'serif' as const,
    rounded: 'normal' as const,
    mono: 'monospace' as const,
  },
  web: {
    sans: 'var(--font-display)' as const,
    serif: 'var(--font-serif)' as const,
    rounded: 'var(--font-rounded)' as const,
    mono: 'var(--font-mono)' as const,
  },
});

export const Typography: Record<string, TextStyle> = {
  /** Main page titles — 32px bold */
  pageTitle: {
    fontSize: 32,
    fontWeight: '800',
    color: '#fff',
    letterSpacing: -0.5,
  },
  /** Large hero text — used in greetings */
  hero: {
    fontSize: 32,
    fontWeight: '800',
    color: '#fff',
    letterSpacing: -0.5,
  },
  /** Section title — 18-20px */
  sectionTitle: {
    fontSize: 20,
    fontWeight: '700',
    color: '#fff',
    letterSpacing: -0.5,
  },
  /** Header title — 20-22px */
  headerTitle: {
    fontSize: 20,
    fontWeight: '700',
    color: '#fff',
  },
  /** Card labels — 16px */
  cardLabel: {
    fontSize: 16,
    fontWeight: '700',
    color: '#fff',
  },
  /** Body text — 16px */
  body: {
    fontSize: 16,
    fontWeight: '500',
    color: '#fff',
  },
  /** Small body — 14px */
  bodySmall: {
    fontSize: 14,
    fontWeight: '500',
  },
  /** Caption — 12-13px */
  caption: {
    fontSize: 13,
    color: '#8892B0',
  },
  /** Large numeric display — balance, etc. */
  numericLarge: {
    fontSize: 36,
    fontWeight: '900',
    color: '#fff',
    letterSpacing: -1,
  },
  /** Button text — 18px bold */
  button: {
    fontSize: 18,
    fontWeight: '700',
    color: '#fff',
    letterSpacing: 0.5,
  },
} as const;

File: guayaba-app/lib/core/theme/typography.dart
import 'package:flutter/material.dart';
import 'package:google_fonts/google_fonts.dart';
import 'colors.dart';

/// Typography presets for the Guayaba/MaaS app.
class AppTypography {
  AppTypography._();

  static TextStyle get pageTitle => GoogleFonts.inter(
        fontSize: 32,
        fontWeight: FontWeight.w800,
        color: AppColors.charcoal,
        letterSpacing: -0.5,
      );

  static TextStyle get hero => GoogleFonts.inter(
        fontSize: 32,
        fontWeight: FontWeight.w800,
        color: AppColors.textPrimary,
        letterSpacing: -0.5,
      );

  static TextStyle get sectionTitle => GoogleFonts.inter(
        fontSize: 20,
        fontWeight: FontWeight.w700,
        color: AppColors.charcoal,
        letterSpacing: -0.5,
      );

  static TextStyle get headerTitle => GoogleFonts.inter(
        fontSize: 18,
        fontWeight: FontWeight.w700,
        color: AppColors.charcoal,
      );

  static TextStyle get cardLabel => GoogleFonts.inter(
        fontSize: 16,
        fontWeight: FontWeight.w700,
        color: AppColors.charcoal,
      );

  static TextStyle get body => GoogleFonts.inter(
        fontSize: 16,
        fontWeight: FontWeight.w500,
        color: AppColors.charcoal,
      );

  static TextStyle get bodySmall => GoogleFonts.inter(
        fontSize: 14,
        fontWeight: FontWeight.w500,
        color: AppColors.charcoal,
      );

  static TextStyle get caption => GoogleFonts.inter(
        fontSize: 13,
        color: AppColors.grayNeutral,
      );

  static TextStyle get numericLarge => GoogleFonts.inter(
        fontSize: 36,
        fontWeight: FontWeight.w900,
        color: AppColors.charcoal,
        letterSpacing: -1,
      );

  static TextStyle get button => GoogleFonts.inter(
        fontSize: 18,
        fontWeight: FontWeight.w700,
        color: Colors.white,
        letterSpacing: 0.5,
      );
}
